extensions/todo-list/src/menu_bar_todo_item.tsx

```diff
@@ -0,0 +1,18 @@
+import { TodoItem, TodoSections } from "./atoms";
+
+import { MenuBarExtra } from "@raycast/api";
+import { useTodo } from "./hooks/useTodo";
+
+export default function MenuBarTodoItem({
+  item,
+  idx,
+  sectionKey,
+}: {
+  item: TodoItem;
+  idx: number;
+  sectionKey: keyof TodoSections;
+}) {
+  const { toggleTodo } = useTodo({ item, idx, sectionKey });
+
+  return <MenuBarExtra.Item title={`${item.completed ? "✓" : "○"}  ${item.title}`} onAction={() => toggleTodo()} />;
```

Comment:
Right now, Raycast isn't guaranteed to wait for a render once an action handler has been executed. Because of this, the list is only re-rendered once you invoke the command from the root search or when you click the item again in the menu bar. We're currently looking into making sure we always perform a render after an action handler, but in the mean time you should be able to work around it by adding an `await setTimeout(1)` in your action handler… 😅

```
import { setTimeout } from "timers/promises";

…

return (
  <MenuBarExtra.Item
    title={`${item.completed ? "✓" : "○"}  ${item.title}`}
    onAction={async () => {
      toggleTodo();
      await setTimeout(1);
    }}
  />
);
``` 

Without await:
https://user-images.githubusercontent.com/1155589/183705350-4fd2e85f-fed5-45a3-ad02-ef807d555916.mov

With await:
https://user-images.githubusercontent.com/1155589/183705901-3f73546a-7638-4620-b095-30ec0df1cc1a.mov